### DISCUSSION TITLE
LifeCycleHook READ operation alternative

### DISCUSSION THREAD

**Discussion by robin-deruyck** [yahoo/elide#3030] [Elide 5]
Hello,

I'm about to upgrade from elide 5 to 6. I want to prepare my current codebase to this migration.
I have seen that READ operation will no longer exist, so I wonder what would be the alternative ? 

I'm using quite a lot of this operation for two actions when a GET request is catched : 
- Some data can be null in my database, it must stay like this for business reasons. So I provide a default value that must not be. persisted. My alternative is using entity converters to do so.

- Some data can be wrong, same problem, business reasons, so I can't clean it myself. I use the READ operation to check data validity, if the check fails, I use another service to generate and persist new values before continuing the "GET" request.

Let me know if you need more informations about my request.
Thx by advance

**Comment by justin-tay** [comment]
I think a similar question was asked before

- https://github.com/yahoo/elide/issues/2815#issuecomment-1272210539

Would that answer work for you?

**Reply by robin-deruyck** [reply]
Thank you, it helps a lot,

I'm gonna try to implement the solutions aklish provided.
